go/vt/schemamanager/schemamanager_test.go

```diff
@@ -249,19 +218,15 @@ func TestSchemaManagerRegisterControllerFactory(t *testing.T) {
 		})
 
 	_, err := GetControllerFactory("unknown")
-	if err == nil || !strings.Contains(err.Error(), "there is no data sourcer factory") {
-		t.Fatalf("controller factory is not registered, GetControllerFactory should return an error, but got: %v", err)
-	}
+	require.ErrorContains(t, err, "there is no data sourcer factory", "controller factory is not registered, GetControllerFactory should return an error")
+
 	_, err = GetControllerFactory("test_controller")
-	if err != nil {
-		t.Fatalf("GetControllerFactory should succeed, but get an error: %v", err)
-	}
+	require.NoError(t, err)
+
 	func() {
 		defer func() {
 			err := recover()
-			if err == nil {
-				t.Fatalf("RegisterControllerFactory should fail, it registers a registered ControllerFactory")
-			}
+			require.NotNil(t, err, "RegisterControllerFactory should fail, it registers a registered ControllerFactory")
```

Comment:
nit, `require.Error` also works here and is probably easier to read

Reply:
`recover` returns a value of type `any`, so i think we can't use `require.Error` here as it requires a value of type `error`.